### PR TITLE
Upserts are tested in a better way

### PR DESCRIPTION
**Pull request by nviennot**
We simply call delete_all on the subscriber collection, it guarantees
that we will upsert on the next update

### CODE REVIEW

File: spec/integration/upsert_spec.rb [hunk 1]
```diff
@@ -39,11 +44,13 @@
 
   context 'when destroying' do
     it 'replicates' do
-      use_null_amqp(:logger_level => Logger::FATAL)
       pub1 = PublisherModel.create(:field_1 => '1', :field_2 => '2', :field_3 => '3')
-      use_real_amqp(:logger_level => Logger::FATAL)
 
-      Promiscuous::Worker.replicate
+      eventually do
+        SubscriberModel.first.should_not == nil
```
Comment (jonstorer) [inline spec/integration/upsert_spec.rb]: I think you've written this eventually do block 3 ways in the pull request. Can you standardize them?

**Comment by kareemk** [comment]
LGTM

**Comment by michaeldauria** [comment]
:+1:

**Comment by yodarjun** [comment]
lgtm

**Pull request closed without merging**